catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/InputTransformerWaiter.java

```diff
@@ -0,0 +1,156 @@
+   */
+  public InputTransformerWaiter(
+      InputTransformerIds inputTransformerIds,
+      List<ServiceReference<InputTransformer>> inputTransformers) {
+    this.inputTransformerIds = inputTransformerIds;
+    this.inputTransformers = inputTransformers;
+    this.transformerCheckPeriodMillis = DEFAULT_TRANSFORMER_CHECK_PERIOD_MILLIS;
+    this.transformerWaitTimeoutMillis = getTransformerWaitTimeout();
+    waitForInputTransformers();
+  }
+
+  @VisibleForTesting
+  InputTransformerWaiter(
+      InputTransformerIds inputTransformerIds,
+      List<ServiceReference<InputTransformer>> inputTransformers,
+      long transformerCheckPeriodMillis,
+      long transformerWaitTimeoutMillis) {
+    this.inputTransformerIds = inputTransformerIds;
+    this.inputTransformers = inputTransformers;
+    this.transformerCheckPeriodMillis = transformerCheckPeriodMillis;
+    this.transformerWaitTimeoutMillis = transformerWaitTimeoutMillis;
+    waitForInputTransformers();
+  }
+
+  /**
+   * Blocks and waits up to a configured ({@code org.codice.ddf.cdm.transformerWaitTimeoutSeconds}
+   * system property) amount of time for the configured {@link InputTransformer} services to be
+   * available.
+   */
+  @SuppressWarnings({
+    "squid:S1481" /* Sonar thinks variables are unused */,
+    "squid:S1854" /* Useless assignment for variable Sonar thinks is unused */
+  })
+  private void waitForInputTransformers() {
+    Set<String> configuredInputTransformers = inputTransformerIds.getIds();
+    if (configuredInputTransformers.isEmpty()) {
+      return;
+    }
+
+    Set<String> requiredInputTransformers = new HashSet<>(configuredInputTransformers);
+    LOGGER.trace(
+        "Beginning to wait for InputTransformers for {} seconds",
+        TimeUnit.MILLISECONDS.toSeconds(transformerWaitTimeoutMillis));
+
+    Failsafe.with(
+            new RetryPolicy()
+                .withMaxDuration(transformerWaitTimeoutMillis, TimeUnit.MILLISECONDS)
+                .retryWhen(false)
+                .withDelay(transformerCheckPeriodMillis, TimeUnit.MILLISECONDS))
+        .withFallback(
+            () -> {
+              throw new IllegalArgumentException(
+                  String.format(
+                      "Not all expected Input Transformers were found within %s seconds. Expected: %s, but didn't find: %s. "
+                          + "CDMs will not be able to start without all required InputTransformers. Either remove the un-found "
+                          + "IDs from the InputTransformer ID JSON files under %s, or ensure the bundles providing "
+                          + "the missing InputTransformers are started.",
+                      TimeUnit.MILLISECONDS.toSeconds(transformerWaitTimeoutMillis),
+                      configuredInputTransformers,
+                      requiredInputTransformers,
+                      inputTransformerIds.getTransformerPath()));
+            })
+        .get(
+            () -> {
```

Comment:
✏️  This code should be moved into a private method and a method reference used here instead. It will make it more readable. Same above